starboard/android/shared/player_set_max_video_input_size.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/android/shared/player_set_max_video_input_size.h"
+
+#include "starboard/extension/player_set_max_video_input_size.h"
+
+#include "starboard/common/log.h"
+#include "starboard/once.h"
+#include "starboard/thread.h"
+
+namespace starboard {
+namespace android {
+namespace shared {
+
+namespace {
+
+// Definitions of any functions included as components in the extension
+// are added here.
+SbOnceControl s_once_flag = SB_ONCE_INITIALIZER;
+SbThreadLocalKey s_thread_local_key = kSbThreadLocalKeyInvalid;
+
+void InitThreadLocalKey() {
+  s_thread_local_key = SbThreadCreateLocalKey(NULL);
+  SB_DCHECK(SbThreadIsValidLocalKey(s_thread_local_key));
+}
+
+void EnsureThreadLocalKeyInited() {
+  SbOnce(&s_once_flag, InitThreadLocalKey);
+  SB_DCHECK(SbThreadIsValidLocalKey(s_thread_local_key));
+}
+
+int GetMaxVideoInputSizeOnCurrentThread() {
+  EnsureThreadLocalKeyInited();
+  // If the key is not valid or there is no value associated
+  // with the key, it would return 0.
+  int value = 0;
+  int* value_ptr = static_cast<int*>(SbThreadGetLocalValue(s_thread_local_key));
+  if (value_ptr != NULL) {
+    value = *value_ptr;
+  }
+  return value;
+}
+
+void SetMaxVideoInputSizeOnCurrentThread(int* max_video_input_size) {
+  SB_DCHECK(max_video_input_size != NULL);
+  InitThreadLocalKey();
```

Comment:
I think we don't need to call InitThreadLocalKey() here, as we have EnsureThreadLocalKeyInited().

Reply:
We only call `SetMaxVideoInputSizeOnCurrentThread()` in sbplayer_bridge.cc, but we need to initialize `s_thread_local_key` before`SbThreadSetLocalValue()`. As a result, I keep `InitThreadLocalKey()` inside `SetMaxVideoInputSizeOnCurrentThread()` to initialize `s_thread_local_key`. Then keeping `EnsureThreadLocalKeyInited()` ensures that it is only called once.

Reply:
InitThreadLocalKey()  creates a new local key to replace the old one, so the existing code will create a new key every time SetMaxVideoInputSizeOnCurrentThread() is called. EnsureThreadLocalKeyInited() runs InitThreadLocalKey() once at the first time executing  SetMaxVideoInputSizeOnCurrentThread(). Can we only create one local key and always use that one?

Reply:
Changed to only `InitThreadLocalKey()` when `s_thread_local_key` is invalid. This can ensure `SetMaxVideoInputSizeOnCurrentThread()` only initializes s_thread_local_key once, and always uses the key.